### ISSUE TITLE
Extract build functionality from Jenkins Helper into Build Repo

### ISSUE THREAD

**Issue by karianna** [AdoptOpenJDK/openjdk-build#1013]
We will need to add a second lib to jenkins that does what we need https://ci.adoptopenjdk.net/configure "Global Pipeline Libraries"
the functions we use are `JobHelper.jobIsRunnable` and `NodeHelper.nodeIsOnline`

**Comment by M-Davies** [comment]
@karianna Is this still needed? I may be wrong but from what I've seen, the [only place where the jobHelper](https://github.com/AdoptOpenJDK/openjdk-build/search?q=jobhelper&unscoped_q=jobhelper) is actually used in the scripts is [here](https://github.com/AdoptOpenJDK/openjdk-build/blob/dab2e5d6199cf12254fa9164e5b924f7944c6246/pipelines/build/common/openjdk_build_pipeline.groovy#L121). 

This could probably be easily replaced with an API call similar to `https://ci.adoptopenjdk.net/view/Test_openjdk/api/json?pretty=true&depth1%22` and then checking the matching test job in `jobs` to see if `buildable` set to `True`: https://ci.adoptopenjdk.net/view/Test_openjdk/job/Test_openjdk14_j9_sanity.openjdk_x86-64_mac/api/json?pretty=true&depth1%22

Which also produces the question, do we need the job helper anymore?

**Comment by karianna** [comment]
> @karianna Is this still needed? I may be wrong but from what I've seen, the [only place where the jobHelper](https://github.com/AdoptOpenJDK/openjdk-build/search?q=jobhelper&unscoped_q=jobhelper) is actually used in the scripts is [here](https://github.com/AdoptOpenJDK/openjdk-build/blob/dab2e5d6199cf12254fa9164e5b924f7944c6246/pipelines/build/common/openjdk_build_pipeline.groovy#L121).
> 
> This could probably be easily replaced with an API call similar to `https://ci.adoptopenjdk.net/view/Test_openjdk/api/json?pretty=true&depth1%22` and then checking the matching test job in `jobs` to see if `buildable` set to `True`: https://ci.adoptopenjdk.net/view/Test_openjdk/job/Test_openjdk14_j9_sanity.openjdk_x86-64_mac/api/json?pretty=true&depth1%22
> 
> Which also produces the question, do we need the job helper anymore?

I agree, if you think you can use that workaround then we could get rid of the job helper.  Hmm, actually might be worth checking the openjdk-tests (and friends) repo.

**Comment by M-Davies** [comment]
Pinging @smlambert and @johnoliver for opinions on scrapping the job helper

**Comment by smlambert** [comment]
@M-Davies - not sure what you are asking... and not sure what JobHelper actually does, but most certainly know that we can run test pipelines without it (at other servers).

**Comment by M-Davies** [comment]
I was more curious if the testing team uses it. I don't want to go through deprecating it if it's still actively used somewhere
